src/ngraph/validation_util.cpp

```diff
@@ -651,221 +651,95 @@ PartialShape ngraph::infer_slice_shape(const Node* node,
-        {
-            if (new_axis.find(i) == new_axis.end())
-            {
-                if (end_dms[i].is_static() && int64_t(end_dms[i]) < 0)
-                {
-                    end_dms[i] = input_shape[j++] + end_dms[i];
-                }
-            }
-            else
-            {
-                end_dms[i] = begin_dms[i];
-            }
-
-            if (end_dms[i].is_dynamic() || begin_dms[i].is_dynamic() || stride_dms[i].is_dynamic())
-            {
-                out_dims.push_back(Dimension::dynamic());
-            }
-            else
-            {
-                out_dims.push_back(static_cast<int64_t>(
-                    ceil(static_cast<float>(abs(int64_t(end_dms[i]) - int64_t(begin_dms[i])) + 1) /
-                         static_cast<float>(abs(int64_t(stride_dms[i]))))));
-            }
-            k = ellipsis_pos1;
-            continue;
-        }
-        stride_dms[i] = (str.size() > sj && str[sj] != 0) ? str[sj++] : 1;
-
-        // Use lower_bounds if mask is not set
-        if (lb_mask.find(j) == lb_mask.end())
-        {
-            if (lb.size() > bj)
-            {
-                begin_dms[i] = lb[bj];
-            }
-            else if (stride_dms[i].is_dynamic())
-            {
-                begin_dms[i] = Dimension::dynamic();
-            }
-            else if (int64_t(stride_dms[i]) > 0)
-            {
-                begin_dms[i] = 0;
-            }
-            else
-            {
-                begin_dms[i] = -1;
-            }
-        }
-        else if (stride_dms[i].is_dynamic())
-        {
-            begin_dms[i] = Dimension::dynamic();
-        }
-        else if (int64_t(stride_dms[i]) > 0)
-        {
-            begin_dms[i] = 0;
-        }
-        else
-        {
-            begin_dms[i] = -1;
-        }
-
-        bj++;
-
-        if (begin_dms[i].is_static() && int64_t(begin_dms[i]) < 0)
-        {
-            begin_dms[i] = input_shape[j] + begin_dms[i];
-        }
-        // Clipping 'begin'
-        if (begin_dms[i].is_static())
-        {
-            if (int64_t(begin_dms[i]) < 0)
-            {
-                begin_dms[i] = 0;
-            }
-            else if (input_shape[j].is_dynamic())
-            {
-                begin_dms[i] = Dimension::dynamic();
-            }
-            else if (int64_t(begin_dms[i]) >= int64_t(input_shape[j]))
+    size_t input_shape_idx = 0;
+    for (size_t axis = 0; axis < begin.size(); ++axis) {
+        if (ellipsis_mask.count(axis)) { // add all dimensions hidden under the ellipsis mask if ellipsis mask is set
+            // only one bit in ellipsis mask is allowed
+            int num_new_axis_after_ellipses = 0;
+            int num_input_axis_before_ellipses = 0;
+            for (size_t i = 0; i < axis; ++i)
             {
-                begin_dms[i] = input_shape[j] - 1;
+                if (!new_axis_mask.count(i))
+                    num_input_axis_before_ellipses++;
```

Comment:
Style: single-statement bodies should be enclosed in `{ }`. Unfortunately `make style-apply` will _not_ fix this one, so please make a pass to update these.